config/packages/prod/monolog.yaml

```diff
@@ -0,0 +1,15 @@
+monolog:
+    handlers:
+        console:
+            channels:
+                - '!event'
+            type: console
+        main:
+            action_level: error
+            excluded_http_codes: [404, 410]
+            handler: nested
+            type: fingers_crossed
+        nested:
+            level: debug
+            path: 'php://stderr'
```

Comment:
`stderr` used not to work well in `fpm` containers because the daemon pollutes the text (e.g. preventing a JSON Lines formatter from working), how does it look like now?

Reply:
You mean about FPM logging in non-JSON and the results being a mix of JSON and not? Still happening...

Reply:
Ah, looks like this is fixed in PHP 7.3 - https://github.com/docker-library/php/pull/725

Reply:
Good to know, there is a Release Candidate 5 for that so it may not be far off (compared to 7.2 which had 6 RCs)

Reply:
Using `php:7.3.0RC5-fpm-alpine` get:

```
app_1  | [2018-11-16 09:20:31] request.INFO: Matched route "ping". {"route":"ping","route_parameters":{"_route":"ping","_controller":"libero.dummy_api.controller.ping"},"request_uri":"http://localhost:8080/ping","method":"GET"} []
web_1  | 172.22.0.1 - - [16/Nov/2018:09:20:31 +0000] "GET /ping HTTP/1.1" 200 14 "-" "curl/7.61.1" "-"
```

Reply:
With 8d3bae2af263e940d0e47d706f861040e13631b1 and `monolog.formatter.json` get:

```
app_1  | {"message":"Some debug message","context":{"some context":"foo"},"level":100,"level_name":"DEBUG","channel":"app","datetime":{"date":"2018-11-16 09:30:13.478482","timezone_type":3,"timezone":"UTC"},"extra":[]}
app_1  | {"message":"Some info message","context":{"some context":"bar"},"level":200,"level_name":"INFO","channel":"app","datetime":{"date":"2018-11-16 09:30:13.478555","timezone_type":3,"timezone":"UTC"},"extra":[]}
app_1  | {"message":"Some notice message","context":[],"level":250,"level_name":"NOTICE","channel":"app","datetime":{"date":"2018-11-16 09:30:13.478582","timezone_type":3,"timezone":"UTC"},"extra":[]}
app_1  | {"message":"Uncaught PHP Exception RuntimeException: \"Some exception\" at /app/src/Controller/ErrorController.php line 28","context":{"exception":{"class":"RuntimeException","message":"Some exception","code":0,"file":"/app/src/Controller/ErrorController.php:28","previous":{"class":"RuntimeException","message":"Some other exception","code":0,"file":"/app/src/Controller/ErrorController.php:28"}}},"level":500,"level_name":"CRITICAL","channel":"request","datetime":{"date":"2018-11-16 09:30:13.488929","timezone_type":3,"timezone":"UTC"},"extra":[]}
```